src/core/services/layout/layout.spec.js

```diff
@@ -132,13 +95,235 @@ describe('layout directives', function() {
       expect(element.hasClass('flex-gt-md-32')).toBe(true);
 
       scope.$apply(function() {
+        // This should be rejected/ignored and the fallback "" value used
         scope.size = "fishCheeks";
       });
 
-      expect(element.hasClass('flex-gt-md-32')).toBe(false);
-      expect(element.hasClass('flex-gt-md-fishCheeks')).toBe(true);
+      expect(element.hasClass('flex-gt-md')).toBe(true);
+      expect(element.hasClass('flex-gt-md-fishCheeks')).toBe(false);
+
+    }));
+
+    testAllSufficesWithValues("flex", allowedValues );
+
+  });
+  describe('using [flex-order] attributes', function() {
+    var flexOrderValues = [
+      -9, -8, -7, -6, -5, -4, -3, -2, -1,
+      0, 1, 2, 3, 4, 5, 6, 7, 8, 9
+    ];
 
+    it('should support attribute without value "<div flex-order>"', inject(function($compile, $rootScope) {
```

Comment:
Since you're adding a bunch of new tests I'll mention this:
You can do 
```js
beforeEach(inject(function($injector) {
  $compile = $injector.get('$compile');
  $rootScope = $injector.get('$rootScope);
});
```

and avoid the boilerplate of having the same `inject` on every test.